src/main/scala/is/hail/variant/Genotype.scala

```diff
@@ -814,6 +818,80 @@ object Genotype {
     gt
   }
 
+  def readBiallelicDosageGenotype(isDosage: Boolean, a: ByteIterator): Double = {
+    val nAlleles = 2
+    val isBiallelic = true
+
+    val flags = a.readULEB128()
+
+    val gt: Int =
+      if (flagHasGT(isBiallelic, flags)) {
+        if (flagStoresGT(isBiallelic, flags))
+          flagGT(isBiallelic, flags)
+        else
+          a.readULEB128()
+      } else
+        -1
+
+    var count = 0
+
+    if (flagHasAD(flags)) {
+      if (flagSimpleAD(flags)) {
+        count += 1
+        val p = Genotype.gtPair(gt)
+        if (p.j != p.k)
+          count += 1
+      } else
+        count += nAlleles
+    }
+
+    if (flagHasDP(flags) && !(flagHasAD(flags) && flagSimpleDP(flags)))
+      count += 1
+
+    a.skipLEB128(count)
+
+    val dosage: Double =
+      if (flagHasPX(flags)) {
+        var px0 = 0
+        var px1 = 0
+        var px2 = 0
+
+        if (gt == 0) {
+          px1 = a.readULEB128()
+          px2 = a.readULEB128()
+          if (isDosage) px0 = 32768 - (px1 + px2)
+        } else if (gt == 1) {
+          px0 = a.readULEB128()
+          px2 = a.readULEB128()
+          if (isDosage) px1 = 32768 - (px0 + px2)
+        } else if (gt == 2) {
+          px0 = a.readULEB128()
+          px1 = a.readULEB128()
+          if (isDosage) px2 = 32768 - (px0 + px1)
+        } else {
+          px0 = a.readULEB128()
+          px1 = a.readULEB128()
+          px2 = a.readULEB128()
+        }
+
+        if (isDosage)
+          (px1 + 2 * px2) * Genotype.dosageNorm
+        else {
+          val p0 = if (px0 < maxPhredInTable) phredToLinearConversionTable(px0) else math.pow(10, px0 / -10.0)
+          val p1 = if (px1 < maxPhredInTable) phredToLinearConversionTable(px1) else math.pow(10, px1 / -10.0)
+          val p2 = if (px2 < maxPhredInTable) phredToLinearConversionTable(px2) else math.pow(10, px2 / -10.0)
+
+          (p1 + 2 * p2) / (p0 + p1 + p2)
```

Comment:
Can you make `phredToBiallelicDosageGenotype` take three arguments and use it here?